### ISSUE TITLE
After Setting uniqueNames property(Y axis) to true chart height increases abnormally

### ISSUE THREAD

**Issue by KumarS-Naveen** [highcharts/highcharts#11796]
#### Expected behaviour
Chart height should remain constant even after after setting UniqueNames property to true as we are not adding anything to Y axis

#### Actual behaviour
After Setting uniqueNames property(Y axis) to true chart height is increases abnormally.

#### Live demo with steps to reproduce
https://jsfiddle.net/dyf20vn3/9/

**Comment by orecus** [comment]
Also experiencing this, seems to behave better with "less" data. The "expand" functionality is also buggy, where you can exapand/hide correctly. The demo above also shows this issue.

**Comment by IAm-Naveen** [comment]
Yes , Also i have raised bugs for expand and collapse issue #11742

**Comment by sebastianbochan** [comment]
Hi @IAm-Naveen,
Thank you for reporting about the problem.

FYI @jon-a-nygaard

**Comment by IAm-Naveen** [comment]
@jon-a-nygaard a work around for this issue will also be a very helpful at this moment

**Comment by KacperMadej** [comment]
As a workaround, you could set chart height to force yAxis cells to have a smaller height.
Demo: https://jsfiddle.net/BlackLabel/9uhdb3z8/

**Comment by IAm-Naveen** [comment]
@KacperMadej Thank you for the solution , but it seems click doesn't work some times after the workaround changes are made.

**Comment by IAm-Naveen** [comment]
@KacperMadej could you please let us know any workaround?

**Comment by Dronh** [comment]
Could you please let me know how to set row height for yAxis?

**Comment by KacperMadej** [comment]
@IAm-Naveen 
That problem is a different issue and it is a reported bug - https://github.com/highcharts/highcharts/issues/11742

@Dronh 
Please see https://github.com/highcharts/highcharts/issues/11796#issuecomment-525832909
For technical support please use https://www.highcharts.com/blog/support/
When contacting the support please remember about adding more info, code context, and a live demo.

**Comment by IAm-Naveen** [comment]
@KacperMadej thanks for that , is there a way to set Y axis each series height so that if the height is increased on setting up UniqueNames we can set height of series as a workaround

**Comment by KacperMadej** [comment]
@IAm-Naveen 
There is no API options that would allow such changes.
You could use `afterSetExtremes` event of the yAxis to recalculate the height of the chart and change it using `chart.setSize`.

**Comment by IAm-Naveen** [comment]
@KacperMadej i will give a shot at this , by the way is there any demo available on how to achieve this ? that would make our life much more easier

**Comment by KacperMadej** [comment]
@IAm-Naveen 
Unfortunately, I am unable to provide a workaround for the case that would be stable in regard to the other Gantt bugs currently open. However, here's a simple POC for a workaround using a wrapper on Axis' `setBreaks`: https://jsfiddle.net/BlackLabel/9uhdb3z8/1/ (wrapper in JS lines 13-33)

**Comment by IAm-Naveen** [comment]
@KacperMadej 
Thanks a ton for the solution and it should be fine as i am having several workaround to tackle issues that i was facing and this was one major hurdle.
I will try out the same , however as i am using angular wrapper will the solution still be same or do i have to tweak a bit

**Comment by KacperMadej** [comment]
No changes in logic are required, but you might be forced to resolve some errors caused by an online sandbox (if you use it) or TS.

If you are working in an online sandbox on an Angular project then you should check if after changes the project is rebuilt or just refreshed, because it might affect how many times the wrapper will be added. To be safe you could add an additional check like in below demo:
https://jsfiddle.net/BlackLabel/9uhdb3z8/2/

TS errors should be easy to resolve by casting to any. If you have an option to create a JS file that won't be checked by TS and will be loaded once after all other Highcharts related files then it would be a good way to go.

**Comment by bm64** [comment]
+1

**Comment by clarghi** [comment]
I also have this problem: I need to have gantt task with subtask, but more than one gantt in a row.
Setting uniqueNames do the trick, but the row grows, and using staticScale the singol gantt graph is very small.
https://jsfiddle.net/clarghi/0egh7846/48/
Thanks!

**Comment by KacperMadej** [comment]
As a workaround, a single series could be used.
An older demo with the workaround: https://jsfiddle.net/BlackLabel/exofhu0w/
The recent demo with the workaround: https://jsfiddle.net/BlackLabel/ry5c3g20/

_Internal note:_
`yAxis.uniqueNames:true` is causing a grouping-like placement of multiple series.

**Comment by Lenaxiao** [comment]
> As a workaround, a single series could be used. An older demo with the workaround: https://jsfiddle.net/BlackLabel/exofhu0w/ The recent demo with the workaround: https://jsfiddle.net/BlackLabel/ry5c3g20/
> 
> _Internal note:_ `yAxis.uniqueNames:true` is causing a grouping-like placement of multiple series.

Will it slow down the rendering by using one single series?

**Comment by raf18seb** [comment]
Hi @Lenaxiao! Sorry for the late reply.

No, it shouldn't have an impact on the rendering time. I'd say it might be even faster with only one series (just an assumption without any tests).

**Comment by bazzacad** [comment]
> The recent demo with the workaround: https://jsfiddle.net/BlackLabel/ry5c3g20/

I'm having this issue too, but I don't see what the workaround is in this demo that's making it work.
Can you point out the line number(s)?

**Comment by bazzacad** [comment]
Oh I get it now, it can only have one series: https://jsfiddle.net/bazzacad/gpedr51y/9/

**Comment by K009** [comment]
Feedback from one of our users: 
-------
Please help me in this as this is very important and urgent for our project and it is impacting our deliverables.

Need only one thing how to make a fixed row height while using yAxis.uniquesNames property to be true.

Note: Not with pure javascript we need it with Angular 9 or greater than 9 version.


We have used below dependencies in our project

     "highcharts": "^9.3.3",
    "highcharts-angular": "2.8.0"



Thanks in advance.
```
